hi-ml/src/health_ml/deep_learning_config.py

```diff
@@ -153,18 +166,52 @@ class WorkflowParams(param.Parameterized):
                      doc="When comparing CSV files during regression tests, use this value as the maximum allowed "
                          "relative difference of actual and expected results. Default: 0.0 (must match exactly)")
     regression_metrics: str = param.String(default=None, doc="A list of names of metrics to compare")
+    run_inference_only: bool = param.Boolean(False, doc="If True, run only inference and skip training after loading"
+                                                        "model weights from the specified checkpoint in "
+                                                        "`src_checkpoint` flag. If False, run training and inference.")
 
     CROSSVAL_INDEX_ARG_NAME = "crossval_index"
     CROSSVAL_COUNT_ARG_NAME = "crossval_count"
 
-    def validate(self) -> None:
-        if self.weights_url and self.local_weights_path:
-            raise ValueError("Cannot specify more than one of local_weights_path, weights_url.")
+    @property
+    def src_checkpoint_is_url(self) -> bool:
+        try:
+            result = urlparse(self.src_checkpoint)
+            return all([result.scheme, result.netloc])
+        except ValueError:
+            return False
+
+    @property
+    def src_checkpoint_is_local_file(self) -> bool:
+        return Path(self.src_checkpoint).exists()
 
+    @property
+    def src_checkpoint_is_aml_run_id(self) -> bool:
+        return re.match(r"[_\w-]*$", self.src_checkpoint.split(":")[0]) is not None
```

Comment:
Should we also include the conditions that `src_checkpoint_is_local_file` and  `src_checkpoint_is_url` be False in this property? Otherwise a URL such as 'https://www.microsoft.com' could satisfy both `src_checkpoint_is_url` and `src_checkpoint_is_aml_run_id`

Reply:
Good point!